torch/_dynamo/variables/functions.py

```diff
@@ -349,6 +360,66 @@ def call_function(
         return super().call_function(tx, args, kwargs)
 
 
+class ContextlibContextManagerFunctionVariable(BaseUserFunctionVariable):
+    """functions that behaves like iterators
+
+    .. note::
+
+        This is only used when the function is annotated with @contextlib.contextmanager
+    """
+
+    def __init__(self, vt: VariableTracker, **kwargs):
+        self.vt = vt
+        self.inline_tracer = None
+
+    def __getattr__(self, name):
+        if name in self.__class__.__dict__.keys():
+            return getattr(self, name)
+        return getattr(self.vt, name)
+
+    # TODO: the __getattr__ call above should have redirect this call to the
+    # underlying variable tracker(?)
+    def export_freevars(self, parent, child):
+        return self.vt.export_freevars(parent, child)
+
+    def call_function(
+        self,
+        tx: "InstructionTranslator",
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        from torch._dynamo.bytecode_transformation import is_generator
+
+        assert is_generator(self.get_code())
+        from torch._dynamo.symbolic_convert import InliningInstructionTranslator
+
+        self.inline_tracer = InliningInstructionTranslator.build_inline_tracer(
+            tx, self, [*self.self_args(), *args], kwargs
+        )
+        # Flag to the tracer to change the behavior of YIELD_VALUE/RETURN_VALUE
+        self.inline_tracer.consume_all_items = False
+
+        return self
+
+    def can_reconstruct(self, tx):
+        # Any graph break should force the entire context manager to run on eager mode
+        return False
+
+    def next_variable(self, tx):
+        from torch._dynamo import exc
+
+        tracer = self.inline_tracer
+
+        try:
+            # inline_call_ has a try/except block that does the same thing
+            # TODO: figure it out why it is not working for this usecase
+            with patch.dict(counters, {"unimplemented": counters["inline_call"]}):
```

Comment:
what's going on here?

Reply:
The `tracer.inline_call` in the symbolic_convert.py has the same `patch.dict` call. This is just to preserve the same behavior.
https://github.com/pytorch/pytorch/blob/3d618019fb5dd1ab20da5b468056a4830d86d323/torch/_dynamo/symbolic_convert.py#L3112-L3115